src/plugins/xtriggers/index.rst

```diff
@@ -0,0 +1,74 @@
+Xtrigger Plugins
+======================================
+
+Xtrigger plugins allow you to install and use xtriggers without them being
+in your ``PYTHONPATH``.
+
+
+Built In Plugins
+----------------
+
+Cylc Flow provides the following xtriggers.
+
+.. autosummary::
+   :toctree: built-in
+   :template: docstring_only.rst
+
+   cylc.flow.xtriggers.echo
+   cylc.flow.xtriggers.wall_clock
+   cylc.flow.xtriggers.workflow_state
+   cylc.flow.xtriggers.xrandom
+
+.. Note: Autosummary generates files in this directory, these are cleaned
+         up by `make clean`.
+
+Developing ``xtrigger`` plugins
+-------------------------------
+
+Cylc uses entry points registered by setuptools to search for xtrigger
+plugins.
+
+Example
+^^^^^^^
+
+In this example a xtrigger plugin is registered.
+
+.. code-block:: python
+   :caption: ``my_package/entry_install.py``
+
+   from my_package.xtriggers import foo, bar
+
+   def xtriggers():
+       # foo contains two xtriggers, 'foo' and 'baz'.
+       # bar contains one xtrigger, 'bar'.
+       return {
+           'foo': foo.foo,
+           'bar': bar.bar,
+           'baz': foo.baz,
+       }
+
+Plugins are registered by registering them with the ``cylc.xtriggers``
+entry points:
+
+.. code-block:: toml
+   :caption: ``setup.cfg``
+
+   [options.entry_points]
+       cylc.xtriggers =
+           my_package = my_package.entry_install:xtriggers
```

Comment:
I'm not keen on using a function to load xtriggers from a package. 

I would recommend registering each xtrigger individually like this:

```diff
    [options.entry_points]
        cylc.xtriggers =
-           my_package = my_package.entry_install:xtriggers
+           foo = my_package.foo:foo
+           bar = my_package.bar:bar
+           baz = my_package.baz:baz
```

The reason for this is that it follows the pattern of existing plugins and reduces the import overheads, i.e. if the baz xtrigger is used in a workflow, only the baz module (and any dependencies) need be imported, whereas loading them from a function, all modules (and dependencies) get imported if any are used.

Reply:
Thanks. I'm learning a lot from this one. I agree your approach is much better and is adopted.